feat: system integrity docs (#29911)
immich-app/immich: docs/docs/administration/system-integrity.md

# System Integrity

## Integrity report

> At a [customizable interval](https://my.immich.app/admin/system-settings?isOpen=integrity-checks), Immich runs integrity checks to ensure that your library is still in tact and there aren't corrupt files.

Reviewer: ```suggestion
At a [customizable interval](https://my.immich.app/admin/system-settings?isOpen=integrity-checks), Immich runs integrity checks to ensure that your library is still intact and there are no corrupt files.
```

There are three kind of issues Immich checks for:

- Untracked files: the path was found in Immich's directories but it is not referenced by Immich

> - Missing files: the path is referenced by Immich but does not actually exist

Reviewer 2: ```suggestion
- Missing files: the path is found in the Immich internal database, but does not exist on disk
```

- Checksum mismatches: the checksum Immich has stored for the file does not match the actual file's checksum anymore

All three run nightly at 3am by default. For the "Checksum files" check, there are additional time and progress limits, as those are the most taxing checks. With these additional limits, Immich can slowly check checksums of your files over the course of a couple of days.

> You can see the results of these checks on the [maintenance](https://my.immich.app/admin/maintenance) page. Here, you can also manually trigger jobs for specific jobs, or all of them. In addition, you can also _refresh_ checks. This will only look at items that have currently been reported on, and check if those have been fixed.

Reviewer 2: ```suggestion
You can see the results of these checks on the [maintenance](https://my.immich.app/admin/maintenance) page. Here, you can also trigger a full scan (a _check_) for specific jobs, or all of them. In addition, you can also _refresh_ checks. This will only look at items that have currently been reported on, and check if those have been fixed.
```

### Common causes

> Most common are untracked files. In many cases those are (corrupted) thumbnails or encoded videos that have been partially been generated at some point and never cleaned up properly. These are usually fine to delete, as both can always be regenerated at a later point. For other files, you need to look at them on a case-by-case basis and check if they already exist in Immich.

Reviewer 2: ```suggestion
Most common are untracked files. In many cases those are corrupted thumbnails or encoded videos that have been partially been generated at some point and never cleaned up properly. These are usually fine to delete, as both can always be regenerated at a later point. For other files, you need to look at them on a case-by-case basis and check if they already exist in Immich.
```

:::info
You might want to run the _missing_ jobs for thumbnails and encoded videos (https://my.immich.app/admin/queues) to make sure all your assets have proper thumbnails and encoded videos. Watch the server logs while running the jobs, in case there are any issues with some assets.
:::

> Missing files are generally worse. If Immich thinks files exist, but they don't actually exist, that's bad. It could be that you just deleted a file on disk from the internal library (don't do that, Immich doesn't support it). It could also be that there are issues with your file storage though. Please carefully investigate missing files, and never hesitate to reach out on [our Discord](https://discord.immich.app) if you have any questions!

Reviewer 2: ```suggestion
Missing files are generally worse. If Immich thinks files exist, but they don't actually exist, that's bad. It could be that you just deleted a file on disk from the internal library (don't do that, Immich doesn't support it). It could also be that there are issues with your file storage. Please carefully investigate missing files, and never hesitate to reach out on [our Discord](https://discord.immich.app) if you have any questions!
```

> Missing files are generally worse. If Immich thinks files exist, but they don't actually exist, that's bad. It could be that you just deleted a file on disk from the internal library (don't do that, Immich doesn't support it). It could also be that there are issues with your file storage though. Please carefully investigate missing files, and never hesitate to reach out on [our Discord](https://discord.immich.app) if you have any questions!

Reviewer 3: `Missing files are generally worse. If Immich thinks files exist, but they don't actually exist, that's bad.`

I think we can leave these subjective comments out of the doc. "Missing files means...", "Checksum mismatches means..." is enough

Author: The goal was to make it clear that people should actually investigate this and not just skim over them.

Reviewer 3: If that is the goal, then you can add an admonition to ask people to make sure to go over the list.

For technical writing, you don't add subjective comments, and usually written in third person. 😄

Author: Pushed some changes @alextran1502

> Checksum mismatches can also be quite bad and indicate file system corruption. It could also be that you previously edited the file in Immich's library in some capacity, which is also not supported and will cause a checksum mismatch. Again, the recommended action is to look at any reported item individually, check it out, try to remember if you changed it or some metadata on it at some point.

Reviewer 2: ```suggestion
Checksum mismatches can also be quite bad and indicative of file system corruption. It could also be that you previously edited a file from the internal library on the disk, which is not supported and will cause a checksum mismatch. Again, the recommended action is to look at any reported item individually, check it out, try to remember if you changed it or some metadata on it at some point. If you edited the file, the supported resolution is to delete the mismatched asset from Immich and reupload it as a new asset.
```
